### PR TITLE
[14.0] [ADD] sale_order_product_assortment_quick

### PR DESCRIPTION
**Pull request by renda-dev**
Bridge module between [sale_quick](https://github.com/OCA/sale-workflow/tree/14.0/sale_quick) and [sale_order_product_assortment](https://github.com/OCA/sale-workflow/tree/14.0/sale_order_product_assortment)

Allows to use feature of sale_quick keeping the domain made in sale_order_product_assortment

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

File: sale_order_product_assortment_quick/models/sale_order.py
Comment (GSLabIt) [inline sale_order_product_assortment_quick/models/sale_order.py]:
```suggestion
       if self.has_allowed_products:
           res = expression.AND([res, [("id", "in", self.allowed_product_ids.ids)]])
        
       return res
```

Comment (GSLabIt) [reply sale_order_product_assortment_quick/models/sale_order.py]: hint ..if u like it, use it

File: sale_order_product_assortment_quick/__manifest__.py
Comment (GSLabIt) [inline sale_order_product_assortment_quick/__manifest__.py:17]: what about to have it auto installing?

Comment (renda-dev) [reply sale_order_product_assortment_quick/__manifest__.py:17]: Totally forgot about that, thanks!

**Review by SilvioC2C — APPROVED** [review]

**Review by GSLabIt — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers I think this bridge module is quite straightforward

cc @CarlosRoca13

**Comment by francesco-ooops** [comment]
@pedrobaeza merge?

File: sale_order_product_assortment_quick/__manifest__.py
Comment (mmequignon) [inline sale_order_product_assortment_quick/__manifest__.py]:
```suggestion
```

File: sale_order_product_assortment_quick/__manifest__.py
Comment (mmequignon) [inline sale_order_product_assortment_quick/__manifest__.py]: I don't think using company-wide account as maintainer is a good idea.
If nobody can use it, so not even you would have merge rights on this module.
On the other hand if any employee have access to this account, everybody would have merge rights.
```suggestion
    "maintainers": ["renda-dev"],
```

File: sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst
Comment (mmequignon) [inline sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst:2]: same here.
```suggestion
* renda-dev <YOUR@EMAIL>
```

**Review by mmequignon — CHANGES_REQUESTED** [review]
A few remarks, regarding the list of maintainers.
I might be wrong though, can someone validate?
The rest looks good.

File: sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst
Comment (francesco-ooops) [reply sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst:2]: hi @mmequignon , I think it's optional to choose whether to publish name and/or email of contributor?

Comment (mmequignon) [reply sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst:2]: Yeah, I just wanted to be sure, as I think it would be a bad idea to give company-wise merge rights on a module.
I chose to temporarily block, as it's more complicated to revert once it's merged :wink:

Comment (mmequignon) [reply sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst:2]: @pedrobaeza could you please confirm?
Thanks!

File: sale_order_product_assortment_quick/__manifest__.py
Comment (francesco-ooops) [reply sale_order_product_assortment_quick/__manifest__.py]: @mmequignon personally I don't see any difference in granting merging rights to a personal account or to a generic company account. Of course, this shouldn't mean this account would be accessible by anyone in the company for a free-for-all mergefest. 

On the other hand, if maintainer dev is on holiday or leaves the company, maintainership of the module is preserved. For example, a dev who's not working with OCA anymore is the maintainer for several modules we commissioned them, and it's a pity those modules are not maintained by anyone as we're not receiving notifications about those modules and maintainer dev is ignoring them.

Also the fact that most modules are pushed by one account but are the result of a team effort should be taken into consideration, IMO :)

Comment (simahawk) [reply sale_order_product_assortment_quick/__manifest__.py]: I don't think these are strong arguments. A maintainer is a specific person.
If you have more persons, add them all.

If you have to address the fact that the person is off or leaving... same: you can easily replace the maintainer or add a new one. 

Plus, I bet no one is going to log in as a bot... which means that this is just a trick to get email notifications -> you can play w/ your own email settings to solve this.

File: sale_order_product_assortment_quick/tests/test_sale_order_product_assortment_quick.py
Comment (simahawk) [inline sale_order_product_assortment_quick/tests/test_sale_order_product_assortment_quick.py]: pls use SavepointCase and move this code to setUpClass + set tracking_disable

```

    @classmethod
    def setUpClass(cls):
        super().setUpClass()
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```

**Review by simahawk — CHANGES_REQUESTED** [review]

File: sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst
Comment (pedrobaeza) [reply sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst:2]: Emails have always been optional (we don't put them, as we don't want spam from leechers asking for free support). I don't have a strong opinion here putting the company or the specific contributor. We put specific contributors as a recognition to them, and in the open source spirit. But it's also something that can go against you. If a weak developer put something in public not in the same quality level than the rest of the company releases, whole company is blamed without distinctions.

File: sale_order_product_assortment_quick/__manifest__.py
Comment (HekkiMelody) [reply sale_order_product_assortment_quick/__manifest__.py]: In our case at least it was not done for email notifications, but truly for the purpose stated of having a company-wide account that e.g. I have access to, but others do not. That way. if I'm absent for a week there are other people in the team (not everyone) that also have access to it and it doesn't simply block everyone else's work.

File: sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst
Comment (francesco-ooops) [reply sale_order_product_assortment_quick/readme/CONTRIBUTORS.rst:2]: Of course nobody has prevented @renda-dev to add their name in the readme file, but if they prefer to remain anonymous I don't see where's the problem

File: sale_order_product_assortment_quick/__manifest__.py
Comment (francesco-ooops) [reply sale_order_product_assortment_quick/__manifest__.py]: > I don't think these are strong arguments. A maintainer is a specific person. If you have more persons, add them all.

TBH I don't see this specified in the guidelines: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/oca_module_lifecycle_maintainer_role.rst

@simahawk I think the maintainer role should take into account both who engineers the module and the dev assigned to build it, as it's not a granted thing that the dev is the best person to ensure that the module is maintained correctly.

As for the generic account use, you can see an example from a few hours ago: https://github.com/OCA/manufacture/pull/1108

So as you see, many modules so far have been merged with both a dev and a generic account as maintainers. If the problem is the absence of a "personal" account, if @renda-dev agrees to be added there's no problem for us :)

Comment (renda-dev) [reply sale_order_product_assortment_quick/__manifest__.py]: No problem for me, I just don't personally like putting my email in the contributors, but I'm fine with being the maintainer for the modules I write.

Anyways, I still prefer that the accessibility of this module would be also kept company-wise, if that's possible.

Comment (francesco-ooops) [reply sale_order_product_assortment_quick/__manifest__.py]: @renda-dev can you also add yourself as maintainer then?

Comment (francesco-ooops) [reply sale_order_product_assortment_quick/__manifest__.py]: @pedrobaeza can we resolve this situation and merge?

Comment (pedrobaeza) [reply sale_order_product_assortment_quick/__manifest__.py]: I'm not the one blocking it...

Comment (simahawk) [reply sale_order_product_assortment_quick/__manifest__.py]: FTR I'm not blocking the PR for this. In fact I'm going to approve here even if it still sound weird.

However, let me explain why all the "blockers" or "impediments" that you think to solve w/ this trick aren't there really. Note that I'm only talking about the "maintainer" role, not the "contributors" entry where you can really put what you want...

> In our case at least it was not done for email notifications, but truly for the purpose stated of having a company-wide account that e.g. I have access to, but others do not. That way. if I'm absent for a week there are other people in the team (not everyone) that also have access to it and it doesn't simply block everyone else's work.

You don't need this trick to "unlock" work  or to "fix" something. To address this issue you must follow a simple rule: always push to a company fork instead of a personal one. Or open another PR if you cannot take over. That's it.

> Anyways, I still prefer that the accessibility of this module would be also kept company-wise, if that's possible.

I don't really get this statement. What prevents you to "access" the module?
These modules are public. Any one inside or outside the company can get their hands on them.

If you have N persons as maintainers behind that bot, just add them all as maintainers.
Is going to be transparent and clear to everyone who they are.

Plus, I don't really see why you think that by not having a maintainer there you will be stuck somehow....
If you want to move fwd w/ a PR that is stalled, you can open another one.
If you don't have rights to merge, you can ask someone else to merge...

I think there are some misunderstandings and I'll be glad if we can talk about this during OCA days :wink:

Comment (francesco-ooops) [reply sale_order_product_assortment_quick/__manifest__.py]: @simahawk @mmequignon please also consider this scenario: junior dev proposes fix to a module, senior dev reviews and merges through company account. I think this is a good practice and reflects the real-case scenario.

Also you can check many modules already merged with company maintainership by [pytech-bot](https://github.com/search?q=org%3AOCA+pytech-bot&type=pullrequests) or [CetmixGitDrone](https://github.com/search?q=org%3AOCA+CetmixGitDrone+manifest&type=code) . 

The advantage for everyone is to ensure module publisher is autonomous merging small bug-fixes and adjustments that are normal after first module publication, without having to bother PSCs all the time.

Is that sufficient to update your review?

**Review by simahawk — APPROVED** [review]

File: sale_order_product_assortment_quick/__manifest__.py
Comment (mmequignon) [reply sale_order_product_assortment_quick/__manifest__.py]: Sorry, I came back late, @simahawk already merged it.

Being in the same company as an author of a given module doesn't means I'm aware about it.

Let's put ourselves in the senior developer shoes here.
I'm contributing on OCA for years maybe, and I'm already getting hundreds or thousands of OCA notifications a week.
This is hard enough to follow, and I wouldn't like (or want) to get more about modules I'm not even aware of.
I'm glad to help, but this is not the way.
If a junior dev needs help about one of its module, there's other means to get me involved in this.

However, if in a given company they're working like that, then why not :shrug: 

Let's consider this case:
- A senior dev from company A creates a complex module.
- Some external guy adds an improvement (with a bug).
- Some junior dev from company A reviews the PR, doesn't catch the bug, and merges it.
That's what I was concerned about.

But if you guys think this is fine, let's go on with it.
Thanks everyone :pray:

Comment (simahawk) [reply sale_order_product_assortment_quick/__manifest__.py]: Regarding the merge itself: in any case, if you are right @mmequignon  and we all agree and we decide that is not fine.... not a reason to block a PR like this. We can roll it back afterwards :wink: 
BTW this is probably a topic for a GH discussion rather than a thread on a PR :wink:

**Pull request merged**
